c/meterpreter/source/extensions/priv/namedpipe_efs.c

```diff
@@ -137,6 +137,7 @@ DWORD WINAPI trigger_efs_connection(LPWSTR pPipeName)
 		if (hr == ERROR_BAD_NETPATH) {
 			dprintf("[ELEVATE] trigger_efs_connection: Success");
 		} else {
+			dwResult = hr;
```

Comment:
According to the docs, it's not clear that [`EfsRpcEncryptFileSrv`](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-efsr/0d599976-758c-4dbd-ac8c-c9db2a922d76) returns a [System Error Code](https://learn.microsoft.com/en-us/windows/win32/debug/system-error-codes--0-499-) which is what dwResult should be.

If it's necessary to set dwResult to an error code, you probably want to map it yourself or set it to an `ERROR_` constant yourself.

Reply:
It can be RPC error code or System Error Code. to make this better, we can do the same on [#L156](https://github.com/rapid7/metasploit-payloads/blob/master/c/meterpreter/source/extensions/priv/namedpipe_efs.c#L156) to print actual error message in debug output and return `RPC_S_CALL_FAILED` anyway. Is that good idea?

Reply:
@NtAlexio2 Alright everything looks good minus this one point. Not sure how people would like to best resolve or if there is any split on the codes here that we might be able to use to our advantage; if someone is able to make a suggestion for the edit I'd be happy to apply it, just not sure that my fix will account for all scenarios.

Once this is done I'd be happy to test this and https://github.com/rapid7/mimikatz/pull/10 and get them landed.

Reply:
Thanks @gwillcox-r7 ! sorry for delay. I was waiting for @smcintyre-r7 answer on this. Fixed on [c9d2e9f](https://github.com/rapid7/metasploit-payloads/pull/625/commits/c9d2e9f9ab20e1ef3b666158c3027068ea566212)

Reply:
Looks good @NtAlexio2 thanks for fixing this!